### PR TITLE
HPCC-15887 Add time attribute to various standard library functions

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

One concern is that this will change CRCs for persists etc I think

**Comment by richardkchapman** [comment]
A quick test suggests that PERSIST crc's are not affected (though I'm not sure why!)

File: plugins/proxies/lib_saltlib.ecllib [hunk 1]
```diff
@@ -22,7 +22,7 @@ EXPORT GeohashDecodeResultRecord := RECORD
     REAL latitude_err;
     REAL longitude_err;
 END;
-export SaltLib := SERVICE : plugin('saltlib'),fold
+export SaltLib := SERVICE : plugin('saltlib'),fold,time
```
Comment (ghalliday) [inline plugins/proxies/lib_saltlib.ecllib]: I'm not sure about these - they extra overhead might be significant.

Comment (richardkchapman) [inline plugins/proxies/lib_saltlib.ecllib]: So might the cumulative impact of all the calls... these functions are not trivial...

If it was the case that a significant amount of time was spent in these functions, we'd want to know...

**Comment by ghalliday** [comment]
I have opened another jira for 6.4 to review the performance - to ensure that it has minimal impact on the query time.

**Pull request merged**